docs/reference/api/media.md

```diff
@@ -0,0 +1,153 @@
+# Media APIs
+
+
+The `@apostrophecms/attachment` module manages all media file upload. Once uploaded, metadata and organization of image and other files are primarily managed through the respective `@apostrophecms/image` and `@apostrophecms/file` piece types. See the [pieces REST API documentation](./pieces.md) for additional information.
+
+## Endpoints
+
+| Method | Path | Description | Auth required |
+|---------|---------|---------|---------|
+|POST | [`/api/v1/@apostrophecms/attachment/upload`](#post-api-v1-apostrophecms-attachment-upload) | Upload a media file | TRUE |
+|POST | [`/api/v1/@apostrophecms/attachment/crop`](#post-api-v1-apostrophecms-attachment-crop) | Crop an image file | TRUE |
+|GET | [`/api/v1/@apostrophecms/image`](#media-get-request) | Get an image document with a specified ID | FALSE |
+|POST | [`/api/v1/@apostrophecms/image`](#media-post-request) | Insert a new image | TRUE |
+|GET | [`/api/v1/@apostrophecms/file`](#media-get-request) | Get a file document with a specified ID | FALSE |
+|POST | [`/api/v1/@apostrophecms/file`](#media-post-request) | Insert a new file | TRUE |
+
+## `POST: /api/v1/@apostrophecms/attachment/upload`
+
+**Authentication required.**
+
+The `body` of the request should be a `FormData` object.
+
+### Request example
+
+```javascript
+// Create a `FormData` object
+const formData = new window.FormData();
+
+// Append a file as per the FormData API:
+// https://developer.mozilla.org/en-US/docs/Web/API/FormData/append
+formData.append('file', myfile);
+
+// Request inside an async function.
+const attachment = await fetch('/api/v1/@apostrophecms/attachment/upload', {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'multipart/form-data'
+  },
+  body: formData
+});
+```
+
+### Response
+
+The successful POST request returns the newly created attachment document.
+
+| Property | Format | Description |
+|----------|------|-------------|
+|`_id` | String | A unique and permanent ID for the document |
+|`_url` | String | File URL, if a non-image file |
```

Comment:
This is always set, even for an image file, in which case I believe it points to the original

Reply:
I'm not getting a `_url` property on an image POST request. Maybe a bug?

Reply:
Could be, yes.